### PR TITLE
Do not segfault on `displaysize(tty)` after `close(tty)`

### LINKED ISSUE #34620: `displaysize(tty)` after `close(tty)` segfaults
It'd be nice if `displaysize(tty)` after `close(tty)` can throw an exception rather than segfault. MWE:

```julia
julia> tty = Base.TTY(fd(Base.Filesystem.open("/proc/self/fd/1", Base.Filesystem.JL_O_WRONLY)))
Base.TTY(RawFD(0x00000016) open, 0 bytes waiting)

julia> displaysize(tty)
(43, 114)

julia> close(tty)

julia> displaysize(tty)

signal (11): Segmentation fault
in expression starting at REPL[4]:1
uv_tty_get_winsize at /workspace/srcdir/libuv/src/unix/tty.c:274
displaysize at ./stream.jl:478
jfptr_displaysize_2966 at /home/takafumi/repos/watch/julia/usr/lib/julia/sys.so (unknown line)
jl_apply at /home/takafumi/repos/watch/julia/src/julia.h:1694 [inlined]
do_call at /home/takafumi/repos/watch/julia/src/interpreter.c:369
eval_value at /home/takafumi/repos/watch/julia/src/interpreter.c:458
eval_stmt_value at /home/takafumi/repos/watch/julia/src/interpreter.c:409 [inlined]
eval_body at /home/takafumi/repos/watch/julia/src/interpreter.c:799
jl_interpret_toplevel_thunk at /home/takafumi/repos/watch/julia/src/interpreter.c:911
jl_toplevel_eval_flex at /home/takafumi/repos/watch/julia/src/toplevel.c:814
jl_toplevel_eval_flex at /home/takafumi/repos/watch/julia/src/toplevel.c:764
jl_toplevel_eval_in at /home/takafumi/repos/watch/julia/src/toplevel.c:843
eval at ./boot.jl:331
eval_user_input at /home/takafumi/repos/watch/julia/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:107
run_backend at /home/takafumi/.julia/packages/Revise/SefCg/src/Revise.jl:1042
#85 at ./task.jl:358
jl_apply at /home/takafumi/repos/watch/julia/src/julia.h:1694 [inlined]
start_task at /home/takafumi/repos/watch/julia/src/task.c:687
unknown function (ip: (nil))
Allocations: 14143307 (Pool: 14139964; Big: 3343); GC: 14
```

### PR DESCRIPTION
**Pull request by tkf**
close #34620

### CODE REVIEW

File: base/stream.jl [hunk 1]
```diff
@@ -455,6 +455,9 @@ displaysize() = (parse(Int, get(ENV, "LINES",   "24")),
                  parse(Int, get(ENV, "COLUMNS", "80")))::Tuple{Int, Int}
 
 function displaysize(io::TTY)
+    # A workaround for #34620 and #26687 (this still has the TOCTOU problem).
+    isopen(io) || throw(ArgumentError("TTY is already closed: $io"))
```
Comment (vtjnash) [inline base/stream.jl]:
```suggestion
    check_open(io)
```
?

**Comment by tkf** [comment]
Tests on Windows failed with

```
Error in testset misc:
Error During Test at D:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\misc.jl:527
  Got exception outside of a @test
  IOError: TTY: bad file descriptor (EBADF)
  Stacktrace:
   [1] uv_error at .\libuv.jl:97 [inlined]
   [2] Base.TTY(::Base.Libc.WindowsRawSocket) at .\stream.jl:193
   [3] (::Main.Test64Main_misc.var"#25#26")(::Base.Libc.WindowsRawSocket, ::Base.PipeEndpoint) at D:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\misc.jl:529
   [4] with_fake_pty(::Main.Test64Main_misc.var"#25#26") at D:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\testhelpers\FakePTYs.jl:58
   [5] top-level scope at D:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\misc.jl:528
```

Does FakePTYs work on Windows? Or am I misusing it?

File: test/misc.jl [hunk 2]
```diff
@@ -522,6 +524,15 @@ if stdout isa Base.TTY
     @test_throws KeyError stdout[:bar]
 end
 
+@testset "`displaysize` on closed TTY #34620" begin
+    Main.FakePTYs.with_fake_pty() do rawfd, _
+        tty = Base.TTY(rawfd)
```
Comment (vtjnash) [inline test/misc.jl:529]:
```suggestion
        tty = open(rawfd)::Base.TTY
```

(though there's also a typo on L292 `s/pipe/io` you'll need to fix)

**Comment by vtjnash** [comment]
This needs a rebase or merge with master to pick up CI fixes

**Pull request merged**